rholang/src/main/scala/coop/rchain/rholang/interpreter/compiler/IndexMapChain.scala

```diff
@@ -0,0 +1,44 @@
+package coop.rchain.rholang.interpreter.compiler
+
+/**
+  *
+  * A structure for keeping track of bound variables. Every time we go under a binding construct
+  * (e.g. match or receive), we add a fresh index map to the top of the chain. For a language like
+  * Java, each index map would represent a method's local variables.
+  *
+  * @param chain A list of bound variable maps.
+  * @tparam T The typing discipline we're enforcing.
+  */
+final case class IndexMapChain[T](chain: Vector[DeBruijnIndexMap[T]]) {
+
+  def get(name: String): Option[IndexContext[T]] = chain.head.get(name)
+
+  def find(name: String): Option[(IndexContext[T], Int)] = {
+    import cats.implicits.catsStdInstancesForVector
+    import cats.Foldable.ops.toAllFoldableOps
```

Comment:
We now have practice to put `import cats.syntax.all._` on top of the file to have all extensions in scope and for the rest of instances just include in local scope like `import cats.instances.vector._`.
This prevents more instance conflicts then just `import cats.implicits._` but it's also less verbose then specifying individual instance.